evm/src/cpu/kernel/asm/core/precompiles/sha256.asm

```diff
@@ -0,0 +1,40 @@
+global precompile_sha256:
+    %stack (address, retdest, new_ctx, kexit_info, gas, address, value, args_offset, args_size, ret_offset, ret_size) ->
+        (new_ctx, kexit_info, ret_offset, ret_size)
+    DUP1
+    SET_CONTEXT
+    // stack: (empty)
+    PUSH 0x100000000 // = 2^32 (is_kernel = true)
+    // stack: kexit_info
+
+    %calldatasize
+    %num_bytes_to_num_words
+    // stack: data_words_len, kexit_info
+    %mul_const(@SHA256_DYNAMIC_GAS)
+    PUSH @SHA256_STATIC_GAS
+    ADD
+    // stack: gas, kexit_info
+    %charge_gas
+
+    // TODO: fix this
+    %zero_out_kernel_general
```

Comment:
@npwardberkeley Tests don't pass without this. Is this expected?  
Also in general, since precompiles are run in their own context, we could just store data in the current context instead of the kernel's context to avoid clobbering.

Reply:
Oh sorry forgot to respond! Yes I think mu code currently does assume that the memory starts with zeroes — lmk if I should change that!